This upstream patch addresses Launchpad bug 1391782. Although it's been
addressed in Kilo, the patch below is still not yet released for Juno.

commit 6859dd61a8472e0cf8f9f4d30f1da7312dc0385d
Author: Joe Gordon <joe.gordon0@gmail.com>
Date:   Fri Nov 14 12:57:35 2014 -0800

    Use oslo db concurrency to generate nova.conf.sample
    
    Make sure we include database and concurrency options in the
    sample config file.
    
    Change-Id: I66e68ec120cba2cee86e36aa26720199876c40a4
    Closes-Bug: #1391782

diff --git a/tools/config/oslo.config.generator.rc b/tools/config/oslo.config.generator.rc
index b6ed64e..afcad61 100644
--- a/tools/config/oslo.config.generator.rc
+++ b/tools/config/oslo.config.generator.rc
@@ -1,2 +1,2 @@
-NOVA_CONFIG_GENERATOR_EXTRA_LIBRARIES=oslo.messaging
+NOVA_CONFIG_GENERATOR_EXTRA_LIBRARIES="oslo.messaging oslo.db oslo.concurrency"
 NOVA_CONFIG_GENERATOR_EXTRA_MODULES=keystonemiddleware.auth_token
